### PR TITLE
SAMBRO: tweet alerts

### PR DESCRIPTION
**Pull request by biplovbhandari**
@flavour Please merge

### CODE REVIEW

**Comment by biplovbhandari** [comment]
@flavour Done :) Please merge

**Comment by biplovbhandari** [comment]
@flavour Done :) Please merge

File: modules/s3cfg.py [hunk 1]
```diff
@@ -2423,6 +2423,13 @@ def get_cap_restrict_fields(self):
         """
 
         return self.cap.get("restrict_fields", False)
+    
+    def get_cap_post_to_twitter(self):
+        """
+            Whether to post the alerts in twitter
+        """
+        
+        return self.cap.get("post_to_twitter", False)
```
Comment (flavour) [inline modules/s3cfg.py]: @ToDo: enhance this by as well as True, being able to specify a specific Twitter channel to tweet on (True => use default channel)

File: modules/templates/SAMBRO/config.py [hunk 2]
```diff
@@ -233,6 +236,49 @@ def onapprove(record):
...
+                try:
+                    import tweepy
+                except ImportError:
+                    current.log.debug("tweepy module needed for sending tweets")
+                else:
+                    T = current.T
+                    db = current.db
+                    atable = s3db.cap_alert
+                    itable = s3db.cap_info
+                    send_tweet = current.msg.send_tweet
```
Comment (flavour) [inline modules/templates/SAMBRO/config.py]: Still have this unnecessary intermediate variable

File: modules/templates/SAMBRO/config.py [hunk 3]
```diff
@@ -233,6 +236,49 @@ def onapprove(record):
...
+                                                            atable.sent,
+                                                            limitby=(0, 1)).first()
+                    # Using English Info Segment for now
+                    # Because tweet has Info Headline
+                    iquery = (itable.alert_id == alert_id) & \
+                             (itable.language == "en-US")
+                    irow = db(iquery).select(itable.headline,
+                                             itable.web,
+                                             limitby=(0, 1)).first()
+                    twitter_text = \
```
Comment (flavour) [inline modules/templates/SAMBRO/config.py]: No @ToDo: on handling the multi-message nicely? ;)

File: modules/s3cfg.py (hunk 1 above)
Comment (biplovbhandari) [inline modules/s3cfg.py]: @flavour If False, is there a deployment setting to assign the channel id?
for tweet, can we do send_tweet(self, text="", recipient=None, channel_id=None) to  send_tweet(self, text="", recipient=None, channel_id=None) or is there other way?

Comment (flavour) [inline modules/s3cfg.py]: Options:
False = No tweeting
True = tweet using default channel
<name of channel>

Currently msg.send_tweet doesn't support that option, so if we get to that requirement then we would need to add it there too (easy to add, just like is already there for msg.poll_twitter)

As I say, this is an @ToDo anyway

**Pull request closed without merging**